cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaVersionTest.java

```diff
@@ -31,6 +31,24 @@ private Reader getKafkaVersionsReader(String kafkaVersions) {
         return new InputStreamReader(KafkaVersion.class.getResourceAsStream("/" + kafkaVersions), StandardCharsets.UTF_8);
     }
 
+    @ParallelTest
+    public void parsingInvalidVersionTest() {
+        KafkaVersion kv = new KafkaVersion("2.8", "2.8-IV1", "2.8-IV1", "3.6.9", false, true, "");
+        assertThat(KafkaVersion.compareDottedIVVersions("2.7-IV1", kv.protocolVersion()), lessThan(0));
+        assertThat(KafkaVersion.compareDottedIVVersions("2.9-IV1", kv.protocolVersion()), greaterThan(0));
+
+        assertThrows(NumberFormatException.class, () -> {
+            KafkaVersion kvFail = new KafkaVersion("why", "you", "little", "3.6.9", false, true, "");
+            KafkaVersion.compareDottedIVVersions("2.8.0", kvFail.protocolVersion());
+        });
+
+        assertThrows(NumberFormatException.class, () -> {
+            KafkaVersion kvFail = new KafkaVersion("2.8.1-IV1", "you", "little", "3.6.9", false, true, "");
+            KafkaVersion.compareDottedVersions("2.8.0", kvFail.protocolVersion());
+        });
```

Comment:
BTW: The `KafkaVersion` instances are managed by us. So they would not contain anything stupid. They will IMHO always be something like `new KafkaVersion("2.8.0", "2.8", "2.8", "3.6.9", false, true, "")`. It is the version you compare it to which will be usually user provided? So maybe the tests make more sense the other way around? Having a valid `KafkaVersion` and comparing it with invalid values?